lib/graphql/execution/lazy/lazy_method_map.rb

```diff
@@ -6,7 +6,14 @@ class Lazy
       # The result of this lookup is cached for future resolutions.
       class LazyMethodMap
         def initialize
-          @storage = {}
+          @storage = Hash.new do |h, value_class|
+            registered_superclass = h.each_key.find { |lazy_class| value_class < lazy_class }
+            if registered_superclass.nil?
+              h[value_class] = nil
+            else
+              h[value_class] = h[registered_superclass]
+            end
```

Comment:
It is a tradeoff, though! Keeping the `.nil?` check reminds the reader that we don't expect `false` here. 

Additionally, this is a _very_ "cold" path. It's run once per class of `resolve` return value, and cached for the lifetime of the program. So, when I applied this change and reran my profiling script, the number of calls to `Kernel#nil?` and `NilClass#nil?` _didn't_ change! (I run one "warmup" query before profiling; perhaps the warmup run had a bit fewer calls.)